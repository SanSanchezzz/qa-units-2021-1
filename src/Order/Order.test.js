import React from 'react'
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import {fakeOrders} from "../data/fakeOrders";
import Order from "./Order";

jest.mock('../utils/getDate');
import {getDate} from '../utils/getDate';

configure({adapter: new Adapter()});

describe('Order.js', () => {
  beforeEach(() => {
    getDate.mockReturnValue('11 января');
  });

  afterEach( () => {
    getDate.mockClear();
  });

  it('render with order is null', () => {
    const wrapper = shallow(<Order
        order = {null}
    />);

    expect(wrapper.getElement()).toBeNull();
  });

  it('getDate not called with null', () => {
    shallow(<Order
        order = {null}
    />);

    expect(getDate).toHaveBeenCalledTimes(0);
  });

  it('render with shop and date are null', () => {
    const wrapper = shallow(<Order
        order = {{shop: null, date: null}}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('getDate not called with shop and date are null', () => {
    shallow(<Order
        order = {{shop: null, date: null}}
    />);

    expect(getDate).toHaveBeenCalledTimes(0);
  });

  it('getDate has been called', () => {
    const order = fakeOrders[0];

    shallow(<Order
        order = {order}
    />);

    expect(getDate).toHaveBeenCalledTimes(1);
  });

  it('render with some fake data', () => {
    const order = fakeOrders[0];

    const wrapper = shallow(<Order
        order = {order}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('getDate has been called without items', () => {
    shallow(<Order
        order = {{shop: 'shop', date: 'date', items: null}}
    />);

    expect(getDate).toHaveBeenCalledTimes(1);
  });

  it('render with items null', () => {
    const wrapper = shallow(<Order
        order = {{shop: 'shop', date: 'date', items: null}}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});

