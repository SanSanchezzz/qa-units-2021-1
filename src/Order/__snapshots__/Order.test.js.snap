// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js getDate has been called 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      11 января
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order.js getDate has been called 2`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      shop
    </span>
    <span
      id="date"
    >
      11 января
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js order is null 1`] = `""`;

exports[`Order.js shop and date are null 1`] = `""`;
